### PR TITLE
Changed to unifiedjs-based markdown parsing

### PR DESCRIPTION
**Pull request by jojanaho**
- kept code structure otherwise intact
- fix for workspace manager test: page-c
  was interpreted as code by the remark parser (due to indentation)
- added vscode debug launch target for unit tests
- note: only tested with foam-workspace-manager unit tests

### CODE REVIEW

File: packages/foam-workspace-manager/test/utils/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+foo
+bar
+`;
+
+describe('WorkspaceManager', () => {
+  it('finds top level headings', () => {
+    const titleA = parseNoteTitleFromMarkdown(pageA);
+    const titleB = parseNoteTitleFromMarkdown(pageB);
+    const titleC = parseNoteTitleFromMarkdown(pageC);
+
```
Comment (riccardoferretti) [inline packages/foam-workspace-manager/test/utils/utils.test.ts:28]: might be worth adding a test for a page that has multiple top level headings?

```
# Page D
hello world

# Another header
hello world
```

Comment (jojanaho) [reply packages/foam-workspace-manager/test/utils/utils.test.ts:37]: Thanks for the suggestion, done :)

**Comment by jevakallio** [comment]
@jojanaho this looks really nice! I tested this locally, and I'm seeing some flaking when running the foam-vscode extension.

Steps I took:

- Clone your PR branch
- `cd packages/foam-workspace-manager && yarn start`
- Open root workspace
- Debug (run VS Code extension)

First run fails with a few syntax errors (sometimes about `import`, other times about the bang operator). It appears that the tests are not correctly being transpiled.

Once I dismiss the errors, the extension runs and works correctly.

I'm not sure if this issue pre-existed, or if it was surfaced by the test code task you added. I'll test a bit more before I merge this.

Example of the error:

```
  ● Test suite failed to run

    SyntaxError: /Users/swanson/dev/foam/packages/foam-workspace-manager/test/WorkspaceManager.test.ts: Unexpected token, expected "," (36:15)

      34 |     const note = ws.getNoteWithLinks('page-a');
      35 |     expect(note).not.toBeNull();
    > 36 |     expect(note!.linkedNotes.map(n => n.id)).toEqual(['page-b', 'page-c']);
         |                ^
      37 |     expect(note!.backlinks.map(n => n.id)).toEqual(['page-b']);
      38 |   });
      39 |

      at Parser._raise (node_modules/@babel/parser/src/parser/error.js:60:45)
      at Parser.raiseWithData (node_modules/@babel/parser/src/parser/error.js:55:17)
      at Parser.raise (node_modules/@babel/parser/src/parser/error.js:39:17)
      at Parser.unexpected (node_modules/@babel/parser/src/parser/util.js:152:16)
      at Parser.expect (node_modules/@babel/parser/src/parser/util.js:129:28)
      at Parser.parseCallExpressionArguments (node_modules/@babel/parser/src/parser/expression.js:817:14)
      at Parser.parseSubscript (node_modules/@babel/parser/src/parser/expression.js:667:31)
      at Parser.parseSubscripts (node_modules/@babel/parser/src/parser/expression.js:576:19)
      at Parser.parseExprSubscripts (node_modules/@babel/parser/src/parser/expression.js:557:17)
      at Parser.parseMaybeUnary (node_modules/@babel/parser/src/parser/expression.js:528:21)

```

**Comment by jojanaho** [comment]
```javascript
   SyntaxError: /Users/swanson/dev/foam/packages/foam-workspace-manager/test/WorkspaceManager.test.ts: Unexpected token, expected "," (36:15)

      34 |     const note = ws.getNoteWithLinks('page-a');
      35 |     expect(note).not.toBeNull();
    > 36 |     expect(note!.linkedNotes.map(n => n.id)).toEqual(['page-b', 'page-c']);
         |                ^
```
Looks funky, almost like there's some old file cached somewhere... Let me know if you find out what causes it, I'm baffled

**Comment by jevakallio** [comment]
I could replicate the flaky VS Code test on master as well, but couldn't pinpoint the source yet, so landing this change and dealing with that separately.

Thank you so much @jojanaho!

**Pull request merged**